Replace config with _config in azure_rm_common.py to support azure-mgmt-network latest version

saito-hideki opened the pull request:
##### SUMMARY

In azure-mgmt-network version 16.0.0 (or later), `config` seems to have been replaced with `_config`. `config` is only used in some logic, but it raises an AttributeError, especially when `cert_validation_mode: ignore` is specified.
This PR replaces `config` with `_config` to address this matter.

- Fixes #903 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- plugins/module_utils/azure_rm_common.py

##### ADDITIONAL INFORMATION
None

saito-hideki committed:
Replace config with _config for the latest azure-mgmt-network.

* Addresses issue #903

Signed-off-by: Hideki Saito <saito@fgrep.org>

Fred-sun commented:
@saito-hideki The Track2 SDK configuration parameter is ‘_config’, and the Track1 configuration parameter is ‘config’. Thank you very much!

saito-hideki commented:
@Fred-sun Thanks for your advice! :)
I'm new to azure python SDK, so I had replaced `config` with `_config` for Track1, but I reverted back. And only replaced it for Track2 like below:

```python
        if not is_track2:
            client.config = self.add_user_agent(client.config)

        if not is_track2 and self.azure_auth._cert_validation_mode == 'ignore':
            client.config.session_configuration_callback = self._validation_ignore_callback
        else:
            client._config.session_configuration_callback = self._validation_ignore_callback
```

Please let me know if I'm wrong about them.

Fred-sun commented:
@saito-hideki The proposal presses the following way to change, can better. Thank you very much!

```e.g.
        if not is_track2:
            client.config = self.add_user_agent(client.config)
            if self.azure_auth._cert_validation_mode == 'ignore':
                client.config.session_configuration_callback = self._validation_ignore_callback
        else:
            if self.azure_auth._cert_validation_mode == 'ignore':
                client._config.session_configuration_callback = self._validation_ignore_callback

saito-hideki commented:
@Fred-sun Thank you for your advice!
I have fixed this PR according to your advice https://github.com/ansible-collections/azure/pull/904#issuecomment-1173464369
I would appreciate it if you could review it :)

xuzhang3 commented:
LGTM
